### PR TITLE
36 get tags service

### PR DESCRIPTION
**Pull request by alexxiaoaida**
Contains both getTags service task and post tags task

### CODE REVIEW

File: ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/controller/TagController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@RequiredArgsConstructor
+public class TagController {
+
+    @Autowired
+    private final TagServiceImpl tagService;
+
+    @PostMapping
+    public ResponseEntity<TagRequestDto> createTag(@RequestBody TagRequestDto tagDto) {
+        Tag createdTag = tagService.createTag(tagDto.getName());
+        TagRequestDto createdTagDto = new TagRequestDto(createdTag.getName());
```
Comment (VoicuVL) [inline ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/controller/TagController.java]: make the conversion in service

File: ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private final TagRepository tagRepository;
+
+    @Override
+    public Tag createTag(String name) {
+        Tag existingTag =  tagRepository.findByName(name);
+        if(existingTag != null){
+            throw new IllegalArgumentException("Tag with name" + name + " already exists");
+        }
+        Tag tag = new Tag(name);
+        return tagRepository.save(tag);
```
Comment (VoicuVL) [inline ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java]: check difference between save and saveAndFlush and use the one you think

File: ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/controller/TagController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return new ResponseEntity<>(createdTagDto, HttpStatus.CREATED);
+    }
+
+    @GetMapping
```
Comment (VoicuVL) [inline ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/controller/TagController.java:32]: you can add @Cacheable

Comment (alexxiaoaida) [reply ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/controller/TagController.java:32]: Am pus Cacheable sub get mapping cred ca de aia nu apare outdated

File: ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        Tag existingTag =  tagRepository.findByName(name);
+        if(existingTag != null){
+            throw new IllegalArgumentException("Tag with name" + name + " already exists");
+        }
+        Tag tag = new Tag(name);
+        tagRepository.saveAndFlush(tag);
+        return convertTagToDto(tag);
+    }
+
+    @CachePut(value="tags")
```
Comment (VoicuVL) [inline ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java:34]: you forgot the override

File: ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    public List<TagRequestDto> getAllTags() {
+        List<Tag> tags = tagRepository.findAll();
+        return convertTagsToDtos(tags);
+    }
+
+
+    //Converting
+    public TagRequestDto convertTagToDto(Tag tag) {
```
Comment (VoicuVL) [inline ness-lab-spring/src/main/java/com/nesslabs/nesslabspring/service/TagServiceImpl.java]: you can make this private if you only use them here and take them out of the interface

**Review by VoicuVL — APPROVED** [review]

**Pull request merged**